src/CommonGrids/CommonGrids.jl

```diff
@@ -0,0 +1,647 @@
+    stretch::Meshes.StretchingRule = Meshes.Uniform(),
+    hypsography_fun = (h_grid, z_grid) -> Grids.Flat(),
+    global_geometry::Geometry.AbstractGlobalGeometry = Geometry.ShallowSphericalGlobalGeometry(
+        radius,
+    ),
+    quad::Quadratures.QuadratureStyle = Quadratures.GLL{n_quad_points}(),
+    h_mesh = Meshes.EquiangularCubedSphere(
+        Domains.SphereDomain{FT}(radius),
+        h_elem,
+    ),
+    h_topology::Topologies.AbstractDistributedTopology = Topologies.Topology2D(
+        context,
+        h_mesh,
+    ),
+    horizontal_layout_type = DataLayouts.IJFH,
+    z_mesh::Meshes.IntervalMesh = DefaultZMesh(
+        FT;
+        z_min,
+        z_max,
+        z_elem,
+        stretch,
+    ),
+    enable_bubble::Bool = false,
+) where {FT}
+    @assert horizontal_layout_type <: DataLayouts.AbstractData
+    @assert ClimaComms.device(context) == device "The given device and context device do not match."
+
+    z_boundary_names = (:bottom, :top)
+    h_grid = Grids.SpectralElementGrid2D(
+        h_topology,
+        quad;
+        horizontal_layout_type,
+        enable_bubble,
+    )
+    z_topology = Topologies.IntervalTopology(context, z_mesh)
+    z_grid = Grids.FiniteDifferenceGrid(z_topology)
+    return Grids.ExtrudedFiniteDifferenceGrid(
+        h_grid,
+        z_grid,
+        hypsography_fun(h_grid, z_grid),
+        global_geometry,
+    )
+end
+
+"""
+    CubedSphereGrid(
+        ::Type{<:AbstractFloat}; # defaults to Float64
+        radius::Real,
+        h_elem::Integer,
+        n_quad_points::Integer,
+        device::ClimaComms.AbstractDevice = ClimaComms.device(),
+        context::ClimaComms.AbstractCommsContext = ClimaComms.context(device),
+        quad::Quadratures.QuadratureStyle = Quadratures.GLL{n_quad_points}(),
+        h_mesh = Meshes.EquiangularCubedSphere(Domains.SphereDomain{FT}(radius), h_elem),
+        h_topology::Topologies.AbstractDistributedTopology = Topologies.Topology2D(context, h_mesh),
+        horizontal_layout_type = DataLayouts.IJFH,
+    )
+
+A convenience constructor, which builds a
+[`Grids.SpectralElementGrid2D`](@ref) given:
+
+ - `FT` the floating-point type (defaults to `Float64`) [`Float32`, `Float64`]
+ - `radius` the radius of the cubed sphere
+ - `h_elem` the number of horizontal elements per side of every panel (6 panels in total)
+ - `n_quad_points` the number of quadrature points per horizontal element
+ - `device` the `ClimaComms.device`
+ - `context` the `ClimaComms.context`
+ - `quad` the quadrature style (defaults to `Quadratures.GLL{n_quad_points}`)
+ - `h_mesh` the horizontal mesh (defaults to `Meshes.EquiangularCubedSphere`)
+ - `h_topology` the horizontal topology (defaults to `Topologies.Topology2D`)
+ - `horizontal_layout_type` the horizontal DataLayout type (defaults to `DataLayouts.IJFH`). This parameter describes how data is arranged in memory. See [`Grids.SpectralElementGrid2D`](@ref) for its use.
```

Comment:
I think I would have preferred `h_layout_type` to be consistent with the other ones, but I think we also use `horizontal_layout_type`, so it impossible to be consistent everywhere.

Reply:
Yeah, I mean, we could use a shorter name here and change it internally, but then we'd have two different names. I'm indifferent. Do you have a preference?

Reply:
No, up to you.

Reply:
I'm going to leave this option verbose since it's more of an internal detail, rather than a grid parameter.